pennylane_lightning/core/src/simulators/lightning_qubit/gates/tests/Test_GateImplementations_Matrix.cpp

```diff
@@ -770,6 +771,49 @@ void testApplyMultiQubitOp() {
     }
 }
 
+template <typename PrecisionT, class GateImplementation>
+void testApplyNQubitOp() {
+    std::mt19937 re{1337};
+    const std::size_t num_qubits = 4;
+    const auto margin = PrecisionT{1e-5};
+
+    DYNAMIC_SECTION("CNOT") {
+        const bool inverse = GENERATE(false, true);
+        const std::size_t wire = GENERATE(0U, 1U, 2U, 3U);
+        const std::size_t control = GENERATE(0U, 1U, 2U, 3U);
+        if (wire == control) {
+            return;
+        }
+        const auto matrix = getPauliX<std::complex, PrecisionT>();
+        auto ref_st = createRandomStateVectorData<PrecisionT>(re, num_qubits);
+        auto st(ref_st);
+
+        GateImplementation::applyNQubitOp(st.data(), num_qubits, matrix.data(),
+                                          {2}, {3}, inverse);
+        GateImplementation::applyCNOT(ref_st.data(), num_qubits, {2, 3},
+                                      inverse);
+
+        REQUIRE(st == approx(ref_st).margin(margin));
+    }
+    DYNAMIC_SECTION("CY") {
+        const bool inverse = GENERATE(false, true);
+        const std::size_t wire = GENERATE(0U, 1U, 2U, 3U);
+        const std::size_t control = GENERATE(0U, 1U, 2U, 3U);
+        if (wire == control) {
+            return;
+        }
+        const auto matrix = getPauliY<std::complex, PrecisionT>();
+        auto ref_st = createRandomStateVectorData<PrecisionT>(re, num_qubits);
+        auto st(ref_st);
+
+        GateImplementation::applyNQubitOp(st.data(), num_qubits, matrix.data(),
+                                          {2}, {3}, inverse);
+        GateImplementation::applyCY(ref_st.data(), num_qubits, {2, 3}, inverse);
+
+        REQUIRE(st == approx(ref_st).margin(margin));
+    }
```

Comment:
Can we also add tests for random unitaries of varying sizes? I'd be more confident if we can ensure it works with 1,2,3 qubit gates, controlled across a multitude of wires. We can likely use the outcome from DQ to create a known set of comparison results.

Reply:
Added some inverse tests.